### PR TITLE
ROE-1655: add implementation to save Historical BO to application ses…

### PR DESCRIPTION
**Pull request by dgolubevs**
…sion object;

### JIRA link
https://companieshouse.atlassian.net/browse/ROE-1655

### Change description
Save to trust to application to session Legal and Individual former beneficial owner.

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/utils/trust/historical.beneficial.owner.mapper.ts [hunk 1]
```diff
@@ -1,24 +1,56 @@
...
+  appData: ApplicationData,
+  trustId: string,
+): Page.TrustHistoricalBeneficialOwner => {
+  const data = getTrustByIdFromApp(appData, trustId);
+
+  return {
+    trustId: data.trust_id,
+    trustName: data.trust_name,
+  };
+};
```
Comment (JoshFen95) [inline src/utils/trust/historical.beneficial.owner.mapper.ts:17]: Can this be removed after the common trust data mapper is added to the main branch?

File: src/model/trust.page.model.ts [hunk 2]
```diff
@@ -17,13 +17,45 @@ type TrustBeneficialOwnerListItem = {
...
   trustName: string;
   boInTrust: TrustBeneficialOwnerListItem[];
 };
 
+type TrustHistoricalBeneficialOwner = {
+  trustId: string;
+  trustName: string;
+};
+
+interface TrustHistoricalBeneficialOwnerFormBasic {
```
Comment (JoshFen95) [inline src/model/trust.page.model.ts:30]: Could we call this `TrustHistoricalBeneficialOwnerFormCommon`

File: views/trust-historical-beneficial-owner.html [hunk 3]
```diff
@@ -15,87 +15,91 @@
...
+          },
           hint: 'For example, 27 3 2007',
-          error: errors.startDate
+          error: errors.ceasedDate
         } %}
         {% include "includes/inputs/date-input.html" %}
 
         {% set fieldParam = {
-          name: 'endDate',
+          name: 'notifiedDate',
```
Comment (JoshFen95) [inline views/trust-historical-beneficial-owner.html:94]: ceasedDate should be used for the "endDate"

notifiedDate should be used for the "startDate"

File: src/utils/trust/historical.beneficial.owner.mapper.ts (hunk 1 above)
Comment (dgolubevs) [reply src/utils/trust/historical.beneficial.owner.mapper.ts:17]: yes, when you implement common mapper it should be replaced with it (test also)

File: src/model/trust.page.model.ts (hunk 2 above)
Comment (dgolubevs) [reply src/model/trust.page.model.ts:30]: fixed

**Review by JoshFen95 — APPROVED** [review]

**Review by okoye-companieshouse — APPROVED** [review]

**Pull request merged**